src/fluree/server/consensus/standalone.clj

```diff
@@ -0,0 +1,113 @@
+
+(defn parse-opts
+  "Extract the opts from the transaction and keywordify the top level keys."
+  [expanded-txn]
+  (let [string-opts (get-first-value expanded-txn const/iri-opts)]
+    (reduce-kv (fn [opts k v]
+                 (assoc opts (keyword k) v))
+               {} string-opts)))
+
+(defn create-ledger!
+  [conn subscriptions watcher {:keys [ledger-id txn opts] :as params}]
+  (go-try
+    (let [create-opts   (parse-opts txn)
+          ledger        (deref!
+                         (fluree/create conn ledger-id create-opts))
+          staged-db     (-> ledger
+                            fluree/db
+                            (fluree/stage txn opts)
+                            deref!)
+          commit-result (deref!
+                         ;; following uses :file-data? and will return map with {:keys [db data-file commit-file]}
+                         (fluree/commit! ledger staged-db {:file-data? true}))
+          broadcast-msg (events/transaction-committed params commit-result)]
+      (broadcast/announce-new-ledger! subscriptions watcher broadcast-msg))))
+
+(defn transact!
+  [conn subscriptions watcher {:keys [ledger-id tx-id txn opts] :as params}]
+  (go-try
+    (let [start-time    (System/currentTimeMillis)
+          _             (log/debug "Starting transaction processing for ledger:" ledger-id
+                                   "with tx-id" tx-id ". Transaction sat in queue for"
+                                   (- start-time (:instant params)) "milliseconds.")
+          ledger        (if (deref! (fluree/exists? conn ledger-id))
+                          (deref! (fluree/load conn ledger-id))
+                          (throw (ex-info "Ledger does not exist" {:ledger ledger-id})))
+          staged-db     (-> ledger
+                            fluree/db
+                            (fluree/stage txn opts)
+                            deref!)
+          commit-result (deref!
+                         (fluree/commit! ledger staged-db {:file-data? true}))
+          broadcast-msg (events/transaction-committed params commit-result)]
+      (broadcast/announce-new-commit! subscriptions watcher broadcast-msg))))
+
+(defn process-event
+  [conn subscriptions watcher event]
+  (go
+    (try
+      (let [[event-type event-msg] event
+
+            result (<! (case event-type
+                         :ledger-create (create-ledger! conn subscriptions watcher event-msg)
+                         :tx-queue      (transact! conn subscriptions watcher event-msg)))]
+        (if (exception? result)
+          (let [error-msg (events/error event-msg result)]
+            (broadcast/announce-error! watcher error-msg))
+          result))
+      (catch Exception e
+        (log/error "Unexpected event message - expected two-tuple of [event-type event-data], "
+                   "and of a supported event type. Received:" event e)))))
+
+(defn new-tx-queue
+  [conn subscriptions watcher]
+  (let [tx-queue (async/chan 16)]
```

Comment:
I set the buffer size to 16 here to allow for 16 pending transactions at once before we report back that the transactor is too busy. I just made that number up because it kinda felt right, but I have no evidence to back this up. I'm happy to make this number higher or lower. Perhaps we could make it a configuration option that users can set.

Reply:
It is probably more size sensitive than quantity sensitive since it sits in memory, and its a lot of work to estimate/maintain the size so I'm not suggesting we go down that path at least for now.

I don't have a better suggestion, mine would be just a guess too. I'd probably guess a number like 50 though, but have no problem with 16 either.

I think ultimately it should be a server config at the point we have more robust server configs.

Reply:
It's easy enough to add it as a config option now with a default.